### PR TITLE
Make the E2E supporting VKS data mover environment.

### PR DESCRIPTION
**Pull request by blackpiglet**
* Add new flag DISABLE_VSPHERE_PLUGIN for E2E test.
* Modify the E2E README for the new parameter.
* Add the VolumeSnapshotClass for VKS.
* Modify the plugin install logic.
* Modify the E2E to support data mover cases in VKS.

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: test/types.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ type VeleroConfig struct {
 	ServiceAccountNameToInstall       string
 	EKSPolicyARN                      string
 	FailFast                          bool
+	DisableVspherePlugin              bool
```
Comment (reasonerjt) [inline test/types.go:121]: Change it to "HasVSpherePlugin"?  And following "if" will only need to check this single flag.

Comment (blackpiglet) [reply test/types.go:121]: I agree that using the positive wording is more intuitive, but we cannot replace all the provider checking with this flag, for example:
``` go
						// If the CloudProvider is vSphere and don't need vSphere plugin,
						// Only install the AWS plugin.
						// If do nothing here, the default behavior is
						// installing both AWS and vSphere plugins.
						if OriginVeleroCfg.CloudProvider == Vsphere &&
							OriginVeleroCfg.DisableVspherePlugin {
							OriginVeleroCfg.Plugins = migrationNeedPlugins[AWS][0]
						}
```

It should be changed to something like the following.
The HasVspherePlugin flag is defaulted to false, so we need to check the provider to make sure it's for vSphere environment.

``` go
						// If HasVspherePlugin is false, only install the AWS plugin.
						// If do nothing here, the default behavior is
						// installing both AWS and vSphere plugins.
						if OriginVeleroCfg.CloudProvider == Vsphere &&
							!OriginVeleroCfg.HasVspherePlugin {
							OriginVeleroCfg.Plugins = migrationNeedPlugins[AWS][0]
						}
```

File: test/e2e/backups/ttl.go [hunk 2]
```diff
@@ -122,19 +122,31 @@ func TTLTest() {
 
 		var snapshotCheckPoint SnapshotCheckPoint
 		if useVolumeSnapshots {
-			if veleroCfg.CloudProvider == Vsphere {
-				// TODO - remove after upload progress monitoring is implemented
+			if veleroCfg.HasVspherePlugin {
```
Comment (reasonerjt) [inline test/e2e/backups/ttl.go:125]: trivial, is the `TODO` no longer valid?

File: test/e2e/backups/ttl.go [hunk 3]
```diff
@@ -188,9 +200,13 @@ func TTLTest() {
...
-				Expect(SnapshotsShouldNotExistInCloud(veleroCfg.CloudProvider,
-					veleroCfg.CloudCredentialsFile, veleroCfg.BSLBucket, veleroCfg.BSLConfig,
-					test.backupName, snapshotCheckPoint)).NotTo(HaveOccurred(), "Fail to get Azure CSI snapshot checkpoint")
+				snapshotCheckPoint.ExpectCount = 0
+				Expect(CheckSnapshotsInProvider(
+					veleroCfg,
+					test.backupName,
+					snapshotCheckPoint,
+					false,
+				)).NotTo(HaveOccurred(), "Fail to get Azure CSI snapshot checkpoint")
```
Comment (reasonerjt) [inline test/e2e/backups/ttl.go:209]: Is the error message correct?

File: test/e2e/bsl-mgmt/deletion.go [hunk 4]
```diff
@@ -205,88 +205,99 @@ func BslDeletionTest(useVolumeSnapshots bool) {
...
-						snapshotCheckPoint, err = GetSnapshotCheckPoint(*veleroCfg.ClientToInstallVelero, veleroCfg, 1, bslDeletionTestNs, backupName_1, []string{podName_1})
+					By(fmt.Sprintf("Snapshot of bsl %s should be created in cloud object store", backupLocation1), func() {
+						snapshotCheckPoint, err = GetSnapshotCheckPoint(
+							*veleroCfg.ClientToInstallVelero,
+							veleroCfg,
+							1,
+							bslDeletionTestNs,
+							backupName1,
+							[]string{podName1},
+						)
```
Comment (reasonerjt) [inline test/e2e/bsl-mgmt/deletion.go:233]: Is the message in L234 correct?

File: test/e2e/backups/ttl.go (hunk 2 above)
Comment (blackpiglet) [reply test/e2e/backups/ttl.go:125]: IMO, the vSphere plugin is not related to the upload progress monitoring.

File: test/e2e/backups/ttl.go (hunk 3 above)
Comment (blackpiglet) [reply test/e2e/backups/ttl.go:209]: No. There are a lot of similar misleading messages.
I will updated them in later PRs.

File: test/e2e/bsl-mgmt/deletion.go (hunk 4 above)
Comment (blackpiglet) [reply test/e2e/bsl-mgmt/deletion.go:233]: No. There are a lot of similar errors. 
I will update them in later PRs.

**Review by reasonerjt — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**